### PR TITLE
refactor(Notification): notification support hidden close button when closeIcon setting to null or false

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [X] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- https://github.com/ant-design/ant-design/discussions/42828

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | notification support hidden close button when closeIcon setting to null or false |
| 🇨🇳 Chinese | 通知组件支持通过设置 closeIcon 为 null 或 false 隐藏关闭按钮 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 08eff49</samp>

This pull request adds a feature to customize the close icon of the notification component and its global configuration method. It updates the documentation, the type definition, the logic, and the test case for the `closeIcon` prop, which can be a boolean, a React node, or undefined. It fixes #30891.

### <samp>🤖 Generated by Copilot at 08eff49</samp>

*  Add a feature to hide the close button of the notification by passing null or false to the `closeIcon` prop (#30891)
  * Update the type definition of the `closeIcon` prop to include boolean in the `ArgsProps` interface ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-c80bd08ac7a1f6a8ae2046fe84cc562807b2d11b1a63316286546b888a86ed08L30-R30))
  * Add a condition to the `TypeIcon` component to return null if the `closeIcon` prop is null or false in `components/notification/PurePanel.tsx` ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccR24))
  * Add the `closeIcon` prop to the destructuring of the `config` object and call the `getCloseIcon` function to get the real close icon in the `useInternalNotification` function in `components/notification/useNotification.tsx` ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL116-R121))
  * Add the `closeIcon` and `closable` props to the `originOpen` function call and pass the `realCloseIcon` as the `closeIcon` prop and set the `closable` prop to true if the `realCloseIcon` is truthy in the `useInternalNotification` function in `components/notification/useNotification.tsx` ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfR137-R138))
  * Add a test case to check that the close button is hidden when the `closeIcon` prop is null or false using the `notification.config` method and four notifications with different values of the `closeIcon` prop ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-ed8976c7e832d9befa4199cc192b025c91cb3ff525a3649834d7545fdd89f690R316-R350))
  * Update the documentation of the `closeIcon` prop in the `API` section and add the boolean type, the default value, and a version note in both files ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9b1700ccbd4da0f60742ae024592c27e0118d7307f5a9ed69a91c560c374749fL51-R51), [link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9954e364a4a4121f77d3f9f87225ef361d39c004720443dd896bb31cfe3c6d22L49-R52))
  * Update the documentation of the `closeIcon` prop in the `API` section for the `notification.config` method and add the same information as the previous change in both files ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9b1700ccbd4da0f60742ae024592c27e0118d7307f5a9ed69a91c560c374749fL71-R71), [link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9954e364a4a4121f77d3f9f87225ef361d39c004720443dd896bb31cfe3c6d22L72-R72))
  * Update the documentation of the `closeIcon` prop in the `notification.config` section and add the same information as the previous change in both files ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9b1700ccbd4da0f60742ae024592c27e0118d7307f5a9ed69a91c560c374749fL102-R102), [link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9954e364a4a4121f77d3f9f87225ef361d39c004720443dd896bb31cfe3c6d22L103-R103))
  * Remove an extra column in the table header of the `API` section in `components/notification/index.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/42927/files?diff=unified&w=0#diff-9954e364a4a4121f77d3f9f87225ef361d39c004720443dd896bb31cfe3c6d22L58-R58))

### CODE REVIEW

File: components/notification/index.zh-CN.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ config 参数如下：
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
 | bottom | 消息从底部弹出时，距离底部的位置，单位像素 | number | 24 |  |
-| closeIcon | 自定义关闭图标 | ReactNode | - |  |
+| closeIcon | 自定义关闭图标 | boolean \| ReactNode | true | 5.7.0：设置为 null 或 false 时隐藏关闭按钮 |
```
[the remark this answers is not in the document]
Comment (kiner-tang) [reply components/notification/index.zh-CN.md:72]: 哪个多余字符？

[the remark this answers is not in the document]
Comment (kiner-tang) [reply components/notification/index.zh-CN.md:72]: 加斜杠是为了转义 | ，防止 md 解析时误解析，不是多余的哈

File: components/menu/MenuItem.tsx [hunk 2]
```diff
@@ -4,11 +4,11 @@ import { Item } from 'rc-menu';
 import toArray from 'rc-util/lib/Children/toArray';
 import omit from 'rc-util/lib/omit';
 import * as React from 'react';
+import { cloneElement, isValidElement } from '../_util/reactNode';
```
Comment (zombieJ) [inline components/menu/MenuItem.tsx:7]: 多了一行不相关的

**Review by zombieJ — APPROVED** [review]

File: components/menu/MenuItem.tsx (hunk 2 above)
Comment (kiner-tang) [reply components/menu/MenuItem.tsx]: prettier 调整的顺序，我重置一下。
